bin/truffleruby

```diff
@@ -120,7 +138,28 @@ if [ -n "$CP" ]; then
 fi
 
 declare -a full_command
-full_command=("$JAVACMD" $JAVA_OPTS "-Xbootclasspath/a:$boot_cp_jar" "${java_args[@]}" org.truffleruby.Main "-Xhome=$root" "-Xlauncher=$root/bin/truffleruby" "${ruby_args[@]}")
+if [ $on_graalvm = false ]; then
+    full_command=(
+        "$JAVACMD"
+        $JAVA_OPTS
+        "-Xbootclasspath/a:$boot_cp_jar"
+        "${java_args[@]}"
+        org.truffleruby.Main
+        "-Xhome=$root"
+        "-Xlauncher=$root/bin/truffleruby"
+        "${ruby_args[@]}"
+    )
+else
+    full_command=(
+        $JAVACMD # no " to let options expand to multiple array elements
+        $JAVA_OPTS
+        "${java_args[@]}"
+        com.oracle.graalvm.Main
+        org.truffleruby.Main
+        "-Xlauncher=$root/bin/truffleruby"
+        "${ruby_args[@]}"
+    )
+fi
```

Comment:
It becomes really hard to follow https://github.com/pitr-ch/truffleruby/blob/1fe6464c828680da1bcb87eb933fbe16a40ad575/bin/truffleruby compared to https://github.com/graalvm/truffleruby/blob/master/bin/truffleruby.
Given this is bash, it's more error-prone and harder to follow any logic.
I ask to keep them split as they share close to nothing.
Moreover, GraalVM-specific stuff (e.g.: `-noverify -polyglot` and paths) should live in GraalVM, not our launcher. It also makes it easier to compared to other launguage launchers.